types/pg/pg-tests.ts

```diff
@@ -156,12 +156,15 @@ const pool = new Pool({
 });
 console.log(pool.totalCount);
 pool.connect((err, client, done) => {
-    if (err) {
+    if (err || client == null) {
```

Comment:
Based on your references, a non-null `err` implies a 'null' `client` and a 'null' `err` implies a non-null `client`.
Is it possible to represent this in the types using overloads, or does the compiler's type inference not pick that up correctly?

This would save us from these kinds of redundant checks.

Reply:
That's likely, but my intention was to make the types more correct. I'm not certain that your assumption is actually the case

Reply:
Hmmm. I feel like there's a point where *usability* is more desirable than *accuracy* though. This current version requires all users to always check both parameters, which feels (and probably is) very redundant.
Perhaps we could ask the maintainers of `pg` if this assumption is true?

Also, I haven't done a lot of Node-things before Promises and async/await were a thing, but isn't the common strategy with these kinds of functions to have two separate callbacks, one for a "result" and one for errors? This doesn't matter for the typings of course, but it'd explain why I've never encountered this before.